cli/0000-cli-improve-ux.md

```diff
@@ -0,0 +1,138 @@
+- Start Date: 2020-05-29
+- RFC PR:
+- Issue:
+
+# Summary
+
+The current CLI was designed seemingly for the internal use and the good UX wasn't among the top priorities in the initial implementation.
+The UX lacks in the number of areas such as consistency, semantics, commands discoverability, no integration with bash/zsh, few options are global when they arguably shouldn't be, no default configs file support, UI that could do a better job helping users to "scan" though the `help` output  
+
+### Goals
+
+- Improve commands discoverability
+- Integrate with Bash/zsh
+- Support default configs file
+- Standardize wording
+- Rework the short versions of commands
+- Bring consistent schema for operations on multiple entities
+- Use hyphens instead of underscores
+- Lower priority: Improve help command output/UI
+
+### Non-goals
+
+- Creating any kind of new CLI business logic
+- Refactoring or improving CLI code base
+
+# Detailed design
+
+### Improve commands discoverability
+
+Problem: As a user i want to easily find commands i'm interested in. Less nesting leads to easier discovery.
+
+Commands discoverability can be improved by limiting CLI commands to a maximum of 2 levels nesting.
+
+Before 
+``` bash
+tctl workflow activity complete {options}
+```
+After 
+``` bash
+tctl activity complete {options}
+```
+
+### Integrate with Bash/zsh
+
+Problem: As a user i will benefit from bash/zsh auto-suggestions, auto-completion and suggestions for mistyped commands.
+
+Before
+
+- Users have to type entire commands and options
+
+After
+
+- Pressing Tab when a command/option is partially typed and there is only one candidate should auto-complete the command/option.
+
+- Pressing Tab when there are multiple candidates should output the candidates and allow to continue typing
+
+- Making a typo in a command/option should output candidates for what user possibly meant to type
+
+### Support default configs file
+
+Problem: As a user i want to be able to change the defaults of some options, such as the default namespace
+Create a default config file that CLI will look into to read them and use in commands.
+If the user passes an option explicitly, ignore the defaults from the config file
+For start, support the following default options through the file:
+ - `--namespace`
```

Comment:
It's not clear what will actually be added and what you're suggesting is nice to have in the future. I assume the only actual change proposed is 

> For start, support the following default options through the file:


But that's hard to grok.